### ISSUE TITLE
2014.7.0 fileserver.update returns error

### ISSUE THREAD

**Issue by mclarkson** [saltstack/salt#17945]
```
# salt-run fileserver.update
[ERROR   ] Git fileserver backend is enabled in master config file, but could not be loaded, are pygit2 and libgit2 installed?
```
But GitPython is installed and remote git repositories for salt and pillar work correctly.

```
# cat /etc/redhat-release 
CentOS release 6.6 (Final)
# salt --versions-report
           Salt: 2014.7.0
         Python: 2.6.6 (r266:84292, Jan 22 2014, 09:42:36)
         Jinja2: unknown
       M2Crypto: 0.20.2
 msgpack-python: 0.1.13
   msgpack-pure: Not Installed
       pycrypto: 2.0.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 3.2.4
           Mako: Not Installed
```

**Comment by notpeter** [comment]
Sorry to reopen, but I'm pretty sure this is a regression in 2014.7.  I use gitfs, but I do not specify any gitfs_provider in my master config.  I'd expect fileserver.update to be able to figure out the correct provider (that salt-master normally uses).

Is there a specific reason that the fileserver salt runner requires gitfs_provider to be set?

**Comment by jfindlay** [comment]
@terminalmage, will you comment here?  I don't know enough about gitfs to answer @notpeter's questions.

**Comment by terminalmage** [comment]
I can't think of a reason why this would be a problem. ``gitfs_provider`` is not a required parameter, and there is logic to determine which provider to use in the event that it is not explicitly defined. I'll have to take a closer look at how the fileserver runner works.

**Comment by terminalmage** [comment]
Fixed in #20854.
